lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -1084,6 +1086,13 @@ def discover_security_groups(group, groups, parent_vpc_id=None, subnet_id=None,
     return list(dict((g['GroupId'], g) for g in found_groups).values())
 
 
+def _supports_ebs_optimized(instance_type):
+    ebs_optimized_types = ['a1', 'c4', 'c5', 'd2', 'f1', 'g3', 'h1', 'i3', 'm4', 'm5', 'p2', 'p3', 'r4', 'r5', 't3', 'u-', 'x1', 'z1']
```

Comment:
This doesn't seem hugely future proof, but the alternative might be a bit harder.

I think the pricing api might be able to help but with significantly more work - ec2instances.info seems to use the API to reasonable effect: https://github.com/powdahound/ec2instances.info/blob/master/ec2.py

Not a blocker though

Reply:
Looking at this again, I can see no way of getting this info from the pricing info.

It seems to be: `(set of instance types with dedicatedEbsThroughput set) - (set of instance types with ebsOptimized set) - {'r2','t3'}`

```
aws pricing get-products --service-code=AmazonEC2 --region us-east-1  --filters Type=TERM_MATCH,Field=operatingSystem,Value=RHEL Type=TERM_MATCH,Field=capacitystatus,Value=AllocatedCapacityReservation Type=TERM_MATCH,Field=location,Value="US East (N. Virginia)" --query PriceList[]  --output text | sed 's/  /^M/g' | grep dedicatedEbsThroughput | while read line; do echo $line | jq -r .product.attributes.instanceType ; done | awk -F. '{print $1}' | sort -u
```

```
aws pricing get-products --service-code=AmazonEC2 --region us-east-1 --filters Type=TERM_MATCH,Field=ebsOptimized,Value=Yes --query PriceList[] --output text | sed 's/  /^M/g' | while read line; do echo $line | jq -r .product.attributes.instanceType ; done | awk -F. '{print $1}' | sort -u
```

Reply:
I'd also prefer not to hardcode this list, but if there's no reasonable alternative I'd be ok with it.

Reply:
One thing to consider with using the pricing API ( or any other API, really ), is that we'd be adding additional API calls to each instance, potentially impacting the performance of the module. 
( Thanks @s-hertel for pointing it out )